torch/csrc/dynamo/python_compiled_autograd.cpp

```diff
@@ -385,7 +385,7 @@ variable_list compiled_autograd(
       // at::getStepCallbacksUnlessEmpty(at::RecordScope::BACKWARD_FUNCTION);
       // if (C10_UNLIKELY(step_callbacks.has_value())) { ... }
 
-      variable_list inputs = input_buffers.lookup(call.node.get()).buffer;
+      variable_list& inputs = input_buffers.lookup(call.node.get()).buffer;
```

Comment:
This change should be safe assuming we only read from input_buffers once per node addr, and there's no duplicate nodes addr

Reply:
it is reassigned to the result of the lifting of tensor_pre_hooks and of pre_hooks, which return cloned inputs bound to the call_hook proxy

without refactoring, we would have up to 3 inputs variable_list and only use one

Reply:
Mutating the stored list seems error-prone.  Could we `std::move()` it to a new variable?